### ISSUE TITLE
Kubernetes resources created with Helm and imported to Pulumi don't do proper diffs

### ISSUE THREAD

**Issue by migueloller** [pulumi/pulumi-kubernetes#924]
When attempting to import existing Kubernetes resources managed by Helm to Pulumi, Pulumi doesn't seem to properly diff the k8s resources. Below are steps to reproduce the issue. I've also deployed the stack using just Pulumi with the exact same programs (i.e., skip `helm install` and just run `pulumi up`) and the diffs are managed correctly. It seems the issue is with importing only.

### Program
```ts
import * as pulumi from "@pulumi/pulumi";
import * as k8s from "@pulumi/kubernetes";

const chart = new k8s.helm.v2.Chart("google", {
  path: "./google",
  transformations: [addImportForHelmChart()],
});

function addImportForHelmChart() {
  return (obj: any, opts: pulumi.CustomResourceOptions): void => {
    if (obj != null && obj.metadata != null && obj.metadata.name) {
      opts.import = obj.metadata.namespace
        ? `${obj.metadata.namespace}/${obj.metadata.name}`
        : obj.metadata.name;
    }
  };
}
```

### Helm Chart
```yaml
// src/google/templates/service.yaml

apiVersion: v1
kind: Service
metadata:
  name: google
  namespace: "{{ .Release.Namespace }}"
spec:
  type: ExternalName
  externalName: www.google.com
```

### Pulumi Preview
```
Previewing update (dev):
+ pulumi:pulumi:Stack: (create)
    [urn=urn:pulumi:dev::pulumi-helm-import-issue::pulumi:pulumi:Stack::pulumi-helm-import-issue-dev]
    + kubernetes:helm.sh/v2:Chart: (create)
        [urn=urn:pulumi:dev::pulumi-helm-import-issue::kubernetes:helm.sh/v2:Chart::google]
warning: inputs to import do not match the existing resource; importing this resource will fail
        = kubernetes:core/v1:Service: (import)
            [id=google]
            [urn=urn:pulumi:dev::pulumi-helm-import-issue::kubernetes:helm.sh/v2:Chart$kubernetes:core/v1:Service::google]
            [provider=urn:pulumi:dev::pulumi-helm-import-issue::pulumi:providers:kubernetes::default_1_4_0::04da6b54-80e4-46f7-96ec-b56ff0331ba9]
          + spec: {
              + externalName: "www.google.com"
              + type        : "ExternalName"
            }
Resources:
    + 2 to create
    = 1 to import
    3 changes
```

### Steps To Reproduce
```
# install Node.js
# install Pulumi
# install minikube
git clone https://github.com/migueloller/pulumi-helm-import-issue
cd pulumi-helm-import-issue
npm i
helm install src/google
pulumi preview --diff
```

**Comment by migueloller** [comment]
> I'm going to move this issue to the k8s provider repo because the issue is with the provider's diff behavior and the lack of the last-applied-configuration annotation on these resources.

Sounds good. Thanks for the info on `enableDryRun`!

**Comment by lblackstone** [comment]
This should be fixed with the `enableServerSideApply` mode available in the [v3.20.1 release](https://github.com/pulumi/pulumi-kubernetes/releases/tag/v3.20.1)
